### ISSUE TITLE
IMPORT : "ERROR: more than one row returned by a subquery used as an expression"

### ISSUE THREAD

**Issue by rimbault** [pgmodeler/pgmodeler#972]
Hi,
I'm trying to import a database schema with pgModeler v0.9.0-beta2 (Build : 20170509) and I have this error: 
`"ERROR: more than one row returned by a subquery used as an expression"`
In my database log : 
`STATEMENT:  CREATE OR REPLACE FUNCTION pg_temp.get_seq_attribs(text,text) RETURNS text AS $$ DECLARE res record; BEGIN EXECUTE 'SELECT start_value, min_value, max_value, increment_by, cache_value, ' || ' CASE WHEN is_cycled IS FALSE THEN NULL ' || ' ELSE is_cycled END FROM ' || '"' || $1 || '"' || '.' || '"' || $2 || '"' || ' LIMIT 1;' INTO res; RETURN replace(replace(res::text,'(','{'),')','}'); END $$ LANGUAGE plpgsql; SELECT sq.oid, sq.relname AS name, sq.relnamespace AS schema, sq.relowner AS owner, (SELECT refobjid || ':' || refobjsubid FROM pg_depend WHERE objid=sq.oid AND deptype='a') AS owner_col, (pg_temp.get_seq_attribs(ns.nspname, sq.relname)) AS attribute, (SELECT description FROM pg_description WHERE objoid = sq.oid AND objsubid = 0) AS comment FROM pg_class AS sq LEFT JOIN pg_namespace AS ns ON ns.oid=sq.relnamespace WHERE relkind='S' AND sq.oid > 12374 AND (sq.oid NOT IN ( 12362,12363,12364,12365,16689,16690,16692,16693,16694,16695,16698,16699,16701)) AND sq.oid IN (17231,72242525,17246,17251,49140123,17259,17261,48467339,60515892,17268,29866241,29866228,17277,17292,17294,17302,38588078,17310,60515708,17318,60515684,60515663,478789,29866215,17329,40848062,17338,1536169,48464033,17237,17361,17367,17372,4321049,17404,17406,17412,29866197,2458303,38587960,38588000,41022742,41022761,60515778,38588046,17456,60515914)`

I can give you more information if needed.
Thanks for this great tool !

**Comment by rimbault** [comment]
I can not reproduce it with a dump / restore.
It' seems that the problem is with this subquery : 
```
(
        SELECT
            refobjid || ':' || refobjsubid
        FROM
            pg_depend
        WHERE
            objid = sq.oid
            AND deptype = 'a') AS owner_col
```
Some objects have 2 "owner_col" : 16683:0 and 17270:1 for exemple
```
xxxxxx=# select * from pg_depend where objid = 17277 and deptype='a';
 classid | objid | objsubid | refclassid | refobjid | refobjsubid | deptype 
---------+-------+----------+------------+----------+-------------+---------
     826 | 17277 |        0 |       2615 |    16683 |           0 | a
    1259 | 17277 |        0 |       1259 |    17270 |           1 | a
(2 rows)
```
One is referenced as pg_class and the other by pg_namespace : 
```
xxxxx=# select relname,reltype from pg_class where oid in(2615,1259);
   relname    | reltype 
--------------+---------
 pg_namespace |   11594
 pg_class     |      83
(2 rows)
```
Hoping it helps !

**Comment by rkhaotix** [comment]
Hi!

Could you pelase provide the dump of your database (without data)?
A similar issue was fixed and is already in the 0.9.0-beta2 (to be released
later this week).

--
Raphael Araújo e Silva
pgModeler Project Developer

raphael@pgmodeler.com.br
http://pgmodeler.com.br
https://github.com/pgmodeler

2017-06-27 4:41 GMT-03:00 rimbault <notifications@github.com>:

**Comment by rimbault** [comment]
I've already try a pg_dump / restore (without data) and no ERROR on with the new database ...
